lorawan-encoding/src/string.rs

```diff
@@ -0,0 +1,168 @@
+use crate::keys::*;
+use crate::parser::*;
+
+#[cfg(feature = "with-to-string")]
+pub extern crate std;
+
+pub use hex::FromHexError;
+
+macro_rules! fixed_len_struct_impl_to_string_msb {
+    (
+        $type:ident,$size:expr;
+    ) => {
+        impl core::str::FromStr for $type {
+            type Err = FromHexError;
+
+            fn from_str(s: &str) -> Result<Self, Self::Err> {
+                let mut res = [0; $size];
+                hex::decode_to_slice(s.as_bytes(), &mut res)?;
+                Ok(Self::from(res))
+            }
+        }
+
+        #[cfg(feature = "with-to-string")]
+        impl std::string::ToString for $type {
+            fn to_string(&self) -> std::string::String {
+                let mut res = std::string::String::with_capacity($size * 2);
+                res.extend(std::iter::repeat('-').take($size * 2));
```

Comment:
Why `-`?
Is there a test with under- and oversized hex string?

Reply:
We never assert size, but the `-` ensure an unwritten char would be obvious.

Reply:
Concerning missized strings, good point! I'll work on that now

Reply:
The hex library handles them nicely it seems